frontend/src/lib/WorkflowParser.ts

```diff
@@ -292,7 +292,7 @@ export default class WorkflowParser {
           outputPaths.push({
             bucket: a.s3!.bucket,
             key: a.s3!.key,
-            source: StorageService.MINIO,
+            source: a.s3!.endpoint === 's3.amazonaws.com' ? StorageService.S3 : StorageService.MINIO,
```

Comment:
I just realized it was in another MR which I have not push yet.

Can u look at this and use this function? It is a more generalized check for any AWS s3 endpoints.

```
/**
 * Check if the provided string is an S3 endpoint (can be any region).
 *
 * @param endpoint minio endpoint to check.
 */
export function isS3Endpoint(endpoint: string = ''): boolean {
  return !!endpoint.match(/s3.{0,}\.amazonaws\.com\.?.{0,}/i);
}
```

And this unit test
```
describe('isS3Endpoint', () => {
    it('checks a valid s3 endpoint', () => {
      expect(isS3Endpoint('s3.amazonaws.com')).toBe(true);
    });

    it('checks a valid s3 regional endpoint', () => {
      expect(isS3Endpoint('s3.dualstack.us-east-1.amazonaws.com')).toBe(true);
    });

    it('checks a valid s3 cn endpoint', () => {
      expect(isS3Endpoint('s3.cn-north-1.amazonaws.com.cn')).toBe(true);
    });

    it('checks an invalid s3 endpoint', () => {
      expect(isS3Endpoint('amazonaws.com')).toBe(false);
    });

    it('checks non-s3 endpoint', () => {
      expect(isS3Endpoint('minio.kubeflow')).toBe(false);
    });
  });

```

Reply:
done